modules/openapi-generator/src/main/resources/Java/libraries/okhttp-gson/ApiClient.mustache

```diff
@@ -1643,21 +1643,25 @@ public class ApiClient {
      * @return The guessed Content-Type
      */
     public String guessContentTypeFromFile(File file) {
-        String contentType = URLConnection.guessContentTypeFromName(file.getName());
```

Comment:
thanks for the PR

can you please add some tests in https://github.com/OpenAPITools/openapi-generator/blob/5a18e9897bd8b2b4a5013a41bccaf93710dd3883/samples/client/petstore/java/okhttp-gson/src/test/java/org/openapitools/client/ApiClientTest.java to test `guessContentTypeFromFile` with different file types to confirm the enhancement is good?

Reply:
I've added them, but it seems that the default implementation for FileTypeDetector relies on file extension (commented test).

I'm not very knowledgeable on java and [the description on how to change the implementation](https://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#probeContentType%28java.nio.file.Path%29), so I'd need some help from someone who is in order to change the implementation to accomodate the commented test, but I'd say that, as long as we use this API, users can at least do it.